internal/incident/sync.go

```diff
@@ -46,7 +46,7 @@ func (i *Incident) AddHistory(historyRow *HistoryRow, fetchId bool) (types.Int,
 	} else {
 		_, err := i.db.NamedExec(stmt, historyRow)
 		if err != nil {
-			return types.Int{}, fmt.Errorf("failed to insert incident history: %w", err)
+			return types.Int{}, err
```

Comment:
If you remove that context information here, please make sure that all callers add sufficient context.

For example, the following code will just log the error without anything saying that the error was related to writing the history:
https://github.com/Icinga/icinga-notifications/blob/f5fb90f73b112104b1a3c975ae8ba414d9b0004e/internal/incident/incident.go#L454-L457

(There may be more callers that need adjustment, I haven't looked through all.)

Reply:
Also affects this code (the error will be logged by the caller with a pretty generic message):
https://github.com/Icinga/icinga-notifications/blob/c4b6db47f6c598bd964f1fff0863dda9bd7faf28/internal/incident/sync.go#L112-L115